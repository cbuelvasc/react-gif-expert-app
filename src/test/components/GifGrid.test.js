import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs')

describe('Test GifGrid component', () => {

    const category = 'One punch';

    let wrapper = shallow(<GifGrid category={category} />);

    beforeEach(() => {
        wrapper = shallow(<GifGrid category={category} />);
    });

    test('Component GifGrid should show correctly', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        expect(wrapper).toMatchSnapshot();
    });

    test('Component GifGrid should show the items with useFetchGifs', () => {
        const gifs = [
            {
                id: 'abc123',
                url: 'https://localhost:3000/any.jpg',
                title: 'Nothing'
            },
            {
                id: 'abc124',
                url: 'https://localhost:3000/anything.jpg',
                title: 'Nothing to see'
            }
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p')).exists().toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
});