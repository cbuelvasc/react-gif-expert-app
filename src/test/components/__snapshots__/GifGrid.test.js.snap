// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifGrid component Component GifGrid should show correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One punch
  </h3>
  <p
    className="animate__animated flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
